src/opm/parser/eclipse/EclipseState/Grid/FieldProps.cpp

```diff
@@ -1211,6 +1215,59 @@ bool FieldProps::tran_active(const std::string& keyword) const {
     return calculator != this->tran.end() && calculator->second.size() > 0;
 }
 
+void FieldProps::apply_numerical_aquifers(const NumericalAquifers& numerical_aquifers) {
+    // TODO: ideally, we should also update the cell_depth, but it is not used for later
+    // in the simulator. So we do not update it here.
+    // Maybe also cell volume? Not sure whether it is used later
+    auto& porv_data = this->init_get<double>("PORV").data;
+    auto& satnum_data = this->int_data["SATNUM"].data;
+    auto& pvtnum_data = this->int_data["PVTNUM"].data;
+
+    const auto& aqu_cell_props = numerical_aquifers.aquiferCellProps();
+    for (const auto& [global_index, cellprop] : aqu_cell_props) {
+        const size_t active_index = this->grid_ptr->activeIndex(global_index);
+        porv_data[active_index] = cellprop.pore_volume;
+        satnum_data[active_index] = cellprop.satnum;
+        pvtnum_data[active_index] = cellprop.pvtnum;
+    }
+
+    this->updateTransWithNumericalAquifer(numerical_aquifers);
+}
+
+void FieldProps::updateTransWithNumericalAquifer(const NumericalAquifers& numerical_aquifers) {
+    const std::array<std::set<size_t>, 3> trans_to_remove = numerical_aquifers.transToRemove(*(this->grid_ptr));
+    std::array<std::vector<Box::cell_index>, 3> index_lists;
+    for (int i = 0; i < 3; ++i) {
+        size_t num = 0;
+        for (const auto& elem : trans_to_remove[i]) {
+            const size_t active_index = this->grid_ptr->activeIndex(elem);
+            index_lists[i].emplace_back(elem, active_index, num);
+            num++;
+        }
+    }
+
+    const std::array<std::string, 3> trans_string {"TRANX", "TRANY", "TRANZ"};
+    for (int i = 0; i < 3; ++i) {
+        const std::string& target_kw = trans_string[i];
+        const std::vector<Box::cell_index>& single_index_list = index_lists[i];
+        auto tran_iter = this->tran.find(target_kw);
+        if (tran_iter == this->tran.end()) {
+            const std::string msg = trans_string[i] + " TranCalculator could not be found when applying numerical aquifer";
+            throw std::logic_error(msg);
+        }
+        assert(tran_iter != this->tran.end());
+        const std::string unique_name = tran_iter->second.next_name();
```

Comment:
OK - this is is insanely complex; and I am sorry to say that we are not quite there yet;-(. The transmissibillity keywords `TRANX, TRANY` and `TRANZ` are treated differently from the all the other keywords; the point is that these keywords are in general calculated in the simulator, so when e.g. implementing
```
MULTIPLY
    TRANX 0.75 /
/
```
we do not have the transmissibility field to start with and don't know what should be mulitiplied with 0.75. Instead the `FieldProps` machinery just stores away the 0.75, and then the simulator calls back to the the `FieldProps::apply()` when the initial transmissibility calculations are complete.

We can tag along with the Tran calculator approach; but ohh my that is complex and will be some work. Maybe you can instead manipulate the `PERMX PERMY` and `PERMZ` arrays - they are treated normally, and if set to zero the simulator will eventually calculate zero transmissibility. What do you think.

Reply:
Let me test it out.  If it works, then it is probably fine. It is simpler than manipulate the transmissiblity.

Reply:
It works. Thanks for the suggestion.

Reply:
With this way (permeability to zero), the probably can happen if someone use `TRAN*` keyword to give a transmissiblity value to these connections.  It was possibly the reason that `TRAN*` manipulation was used back then, while I do not remember exactly now.  Let hope we do not have that kind of case. 

Or, we check whether the `TRAN*` keyword is specified related to a numerical aquifer cells when parsing `TRAN*` keyword.

Reply:
There is always a possible twist. Maybe the *best* solution is to implement this - i.e. set TRAN to zero for the aquifer cells - in `ecltransmissibility.hh` in opm-simulators?

Reply:
It is possible, between these two function calls. 
```c++
        applyEditNncToGridTrans_(globalToLocal);
        applyNncToGridTrans_(globalToLocal);
```

Reply:
I have no strong feelings here; at least for me it is much simpler to work in opm-common.

Reply:
Yes. Let us have such a case first, then we tackle it.